app/models/job_proxy_dispatcher.rb

```diff
@@ -144,11 +174,12 @@ def assign_proxy_to_job(proxy, job)
     end
   end
 
-  def pending_jobs
+  def pending_jobs(containers = false)
     @zone = MiqServer.my_zone
     Job.order(:id)
       .where(:state           => "waiting_to_start")
       .where(:dispatch_status => "pending")
+      .where("target_class #{containers ? "=" : "!="} ?", ContainerImage.name)
```

Comment:
It looks like we set target_class to "VmOrTemplate " for vms.  Maybe callers can pass their class:

`pending_jobs(ContainerImage)`
`pending_jobs(VmOrTemplate)`

`active_scans(ContainerImage)`
`active_scans(VmOrTemplate)`

These methods can call klass.base_class.name to ensure we get the base class's name.

That would eliminate some of this complexity.

It should be pretty straightforward to verify this doesn't break JPD for vms.